docs/params.rst

```diff
@@ -0,0 +1,119 @@
+==========
+Parameters
+==========
+
+:ref:`Spider arguments <spiderargs>` have some limitations:
+
+-   There is no standard way for spiders to indicate which parameters they
+    expect or support.
+
+-   Since arguments can come from the command line, and those are always
+    strings, you have to write code to convert arguments to a different type
+    when needed.
+
+-   If you want argument validation, such as making sure that arguments are of
+    the right type, or that required arguments are present, you must implement
+    validation yourself.
+
+scrapy-spider-metadata allows overcoming those limitations.
+
+.. _define-params:
+
+Defining supported parameters
+=============================
+
+To define spider parameters, define a subclass of `pydantic.BaseModel`_,
+and then make your spider also inherit from
+:class:`~scrapy_spider_metadata.Args` with your parameter specification class
+as its parameter:
+
+.. _pydantic.BaseModel: https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel
+
+.. code-block:: python
+
+    from pydantic import BaseModel
+    from scrapy import Request, Spider
+    from scrapy_spider_metadata import Args
+
+    class MyParams(BaseModel):
+        pages: int
+
+    class MySpider(Args[MyParams], Spider):
+        name = "my_spider"
+
+        def start_requests(self):
+            for index in range(1, self.args.pages + 1):
+                yield Request(f"https://books.toscrape.com/catalogue/page-{index}.html")
+
+
+To learn how to define parameters in your `pydantic.BaseModel`_
+subclass, see the `Pydantic usage documentation
+<https://docs.pydantic.dev/latest/usage/models/>`_.
+
+.. _define-params-limits:
+
+However, scrapy-spider-metadata only supports a subset of Pydantic features:
+
+-   Only the following parameter types are supported: :class:`str`,
+    :class:`int`, :class:`float`, :class:`bool`, :class:`~enum.Enum`,
+    :class:`~enum.IntEnum`.
+
+-   Defining multiple types for a given parameter is not supported.
+
+Using other Pydantic features can prevent scrapy-spider-metadata from working
+as expected.
+
+.. note:: If you wish to use a Pydantic feature not listed above, please see if
+    it has already been requests in `our issue tracker
+    <https://github.com/scrapy-plugins/scrapy-spider-metadata/issues>`_, and if
+    not, feel free to request it.
```

Comment:
What do you think about moving this section later, after explaining how to use the arguments?

I also wonder what is going to work, and what won't work. It seems the limitations apply to the jsonschema export (and Scrapy Cloud support), but otherwise it should work. So, if some user just wants to parse arguments in an easier way, it can be ok for them to use more pydantic features. 

Is the idea to promote the happy path where all features work?

Reply:
https://github.com/scrapy-plugins/scrapy-spider-metadata/pull/1/commits/459a3f65d3d3b5bfe9ee62670b3bec60cf7c5d35 ?